ui/framework/src/ui-framework/frontstage/ModalSettingsStage.tsx

```diff
@@ -0,0 +1,93 @@
+/** @packageDocumentation
+ * @module Frontstage
+ */
+
+import "./ModalSettingsStage.scss";
+import * as React from "react";
+import { BackstageItemUtilities, ConditionalBooleanValue, IconSpecUtilities, StageUsage } from "@bentley/ui-abstract";
+import settingsIconSvg from "@bentley/icons-generic/icons/settings.svg?sprite";
+import { IModelApp, NotifyMessageDetails, OutputMessagePriority, OutputMessageType } from "@bentley/imodeljs-frontend";
+import { Logger } from "@bentley/bentleyjs-core";
+import { Centered, SettingsContainer } from "@bentley/ui-core";
+import { FrontstageManager, ModalFrontstageInfo, ModalFrontstageRequestedCloseEventArgs } from "./FrontstageManager";
+import { UiFramework } from "../UiFramework";
+import { SyncUiEventId } from "../syncui/SyncUiEventDispatcher";
+
+function ModalSettingsStage({initialSettingsTabId}: {initialSettingsTabId?: string}) {
+  const id=FrontstageManager.activeFrontstageDef?.id??"none";
+  const stageUsage=FrontstageManager.activeFrontstageDef?.usage??StageUsage.General;
+  const tabEntries = UiFramework.settingsManager.getSettingEntries(id, stageUsage);
+  const noSettingsAvailableLabel = React.useRef(UiFramework.i18n.translate("UiFramework:settings.noSettingsAvailable"));
+
+  const currentSettingsTab = React.useCallback(() => {
+    const categoryToFind = initialSettingsTabId ? initialSettingsTabId.toLowerCase() : tabEntries[0].tabId.toLowerCase();
+    let foundTab = tabEntries.find((entry) => entry.tabId.toLowerCase() === categoryToFind);
+    if (!foundTab)
+      foundTab = tabEntries.find((entry) => entry.label.toLowerCase() === categoryToFind);
+    return foundTab;
+  },[initialSettingsTabId, tabEntries]);
+
+  React.useEffect (()=>{
+    const handleFrontstageCloseRequested = ({modalFrontstage, stageCloseFunc}: ModalFrontstageRequestedCloseEventArgs) => {
+      // istanbul ignore else
+      if (modalFrontstage instanceof SettingsModalFrontstage && stageCloseFunc) {
+        UiFramework.settingsManager.closeSettingsContainer (stageCloseFunc);
+      }
+    };
+    return FrontstageManager.onCloseModalFrontstageRequestedEvent.addListener(handleFrontstageCloseRequested);
+  }, [tabEntries]);
+
+  return (
+    <div className="uifw-settings-container">
+      {tabEntries.length ?
+        <SettingsContainer tabs={tabEntries} currentSettingsTab={currentSettingsTab()} settingsManager={UiFramework.settingsManager} /> :
+        <Centered>{noSettingsAvailableLabel.current}</Centered>
+      }
+    </div>
+  );
+}
+
+/** Modal frontstage displaying and editing settings from registered settings providers. See [SettingsManager]($ui-core)
+ * and [SettingsContainer]($ui-core).
+ * @beta
+ */
+export class SettingsModalFrontstage implements ModalFrontstageInfo {
+  public static id = "ui-framework.modalSettingsStage";
+  public title: string = UiFramework.i18n.translate("UiFramework:settings.settingsStageLabel");

```

Comment:
Can use UiFramework.translate in this file instead of  UiFramework.i18n.translate, and also remove the "UiFramework:" namespace when using UiFramework.translate.